### PR TITLE
Port small LACP functions. Add testing using temp netns spaces.

### PR DESCRIPTION
**Pull request by colinmitchell-ggl**
This will help clear out some duplicated bonding code from cmds.
Tests require root, but should be fine outside QEMU.

Signed-off-by: Colin Mitchell <colinmitchell@google.com>

### CODE REVIEW

**Review by rminnich — APPROVED** [review]
This is pretty slick!

Note that you are missing a  bit of test coverage, and in future, that will cause PRs like this to be blocked. But IIUC you need this so ...

**Comment by rjoleary** [comment]
The tests which are failing look unrelated 😞  . I'm going to try merging in latest main branch.

```
=== RUN   TestGetOne
    get_test.go:34: 
        	Error Trace:	get_test.go:34
        	Error:      	Not equal: 
        	            	expected: "firmware_version(RO) => 1.2.3\n\nfirmware_version(RW) => 3.2.1\n\n"
        	            	actual  : "No variable named 'firmware_version' found\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,5 +1,2 @@
        	            	-firmware_version(RO) => 1.2.3
        	            	+No variable named 'firmware_version' found
        	            	 
        	            	-firmware_version(RW) => 3.2.1
        	            	-
        	            	-
        	Test:       	TestGetOne
--- FAIL: TestGetOne (0.00s)
=== RUN   TestGetAll
    get_test.go:46: 
        	Error Trace:	get_test.go:46
        	Error:      	Not equal: 
        	            	expected: "firmware_version(RO) => 1.2.3\n\nsomething(RO) => else\n\nfirmware_version(RW) => 3.2.1\n\n"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,7 +1 @@
        	            	-firmware_version(RO) => 1.2.3
        	            	 
        	            	-something(RO) => else
        	            	-
        	            	-firmware_version(RW) => 3.2.1
        	            	-
        	            	-
        	Test:       	TestGetAll
--- FAIL: TestGetAll (0.00s)
```

File: pkg/lacp/lacp.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+// Package lacp provides methods to setup and teardown linux LACP bonds.
+package lacp
+
+import (
+	"fmt"
+
+	"github.com/vishvananda/netlink"
+)
+
+// Follow netbase's bonding settings found in gbuild/netbase/etc/kernel/%uname-r%/netbase/libbond.sh
```
Comment (rjoleary) [inline pkg/lacp/lacp.go:14]: Is there a URL that can be linked for reference?

File: pkg/lacp/lacp_linux_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	origns, _ := netns.Get()
+
+	// new temporary namespace so we don't pollute the host
+	// lock thread since the namespace is thread local
+	ns, err := netns.New()
+	if err != nil {
+		t.Fatal("Failed to create newns", err)
+	}
+	err = netns.Set(ns)
+	if err != nil {
```
Comment (rjoleary) [inline pkg/lacp/lacp_linux_test.go:41]: `if err := netns.Set(ns); err != nil {`

File: pkg/lacp/lacp_linux_test.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	defer tearDown()
+	links := setupDefaultLinks(t)
+
+	bondName := "urBond"
+	bond := netlink.NewLinkBond(netlink.LinkAttrs{
+		Name:         bondName,
+		HardwareAddr: mac(t, "AA:BB:CC:DD:EE:FF"),
+	})
+	err := netlink.LinkAdd(bond)
+	if err != nil {
```
Comment (rjoleary) [inline pkg/lacp/lacp_linux_test.go:135]: if err := ...; err != nil {

File: pkg/lacp/lacp_linux_test.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+		t.Fatalf("Failed adding bond %v: %v", bond, err)
+	}
+
+	err = RemoveExistingBonds()
+	if err != nil {
```
Comment (rjoleary) [inline pkg/lacp/lacp_linux_test.go:140]: if err := ...; err != nil {

File: pkg/lacp/lacp_linux_test.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+		t.Fatalf("Failed RemoveExistingBonds. Got %v, want: nil", err)
```
Comment (rjoleary) [inline pkg/lacp/lacp_linux_test.go:141]:
```suggestion
		t.Fatalf("RemoveExistingBonds() = %v, want nil", err)
```

File: pkg/lacp/lacp_linux_test.go [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	got, err := netlink.LinkByName(bondName)
+	if err == nil {
+		t.Errorf("Found bond with name %s: Got: %v; Want nil", got.Attrs().Name, got)
```
Comment (rjoleary) [inline pkg/lacp/lacp_linux_test.go:146]:
```suggestion
		t.Errorf("Found bond with name %s: Got: %v; Want no bond with that name", got.Attrs().Name, got)
```

File: pkg/lacp/lacp_linux_test.go [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+func TestCreateLACPBond(t *testing.T) {
+	tearDown := setupNetlinkTest(t)
+	defer tearDown()
+	links := setupDefaultLinks(t)
+
+	bondName := "urBond"
+	// Expected to use array[0] for MAC
+	wantBondMac := links[0].Attrs().HardwareAddr
+	_, err := CreateLACPBond([]netlink.Link{links[0], links[1]}, bondName)
+	if err != nil {
```
Comment (rjoleary) [inline pkg/lacp/lacp_linux_test.go:163]: if _, err := ...; err != nil {

**Pull request closed without merging**